crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -116,33 +121,27 @@ impl ProtocolFees {
             buy: quote.buy_amount,
             fee: quote.fee,
         };
-        let protocol_fees = if self
-            .protocol_fee_exempt_addresses
-            .contains(&order.metadata.owner)
-        {
-            vec![]
-        } else {
-            self
-                    .fee_policies
-                    .iter()
-                    // TODO: support multiple fee policies
-                    .find_map(|fee_policy| {
-                        let outside_market_price = boundary::is_order_outside_market_price(&order_, &quote_, order.data.kind);
-                        match (outside_market_price, &fee_policy.order_class) {
-                            (_, OrderClass::Any) => Some(&fee_policy.policy),
-                            (true, OrderClass::Limit) => Some(&fee_policy.policy),
-                            (false, OrderClass::Market) => Some(&fee_policy.policy),
-                            _ => None,
-                        }
-                    })
-                    .and_then(|policy| match policy {
-                        policy::Policy::Surplus(variant) => variant.apply(&order),
-                        policy::Policy::PriceImprovement(variant) => variant.apply(&order, quote),
-                        policy::Policy::Volume(variant) => variant.apply(&order),
-                    })
-                    .into_iter()
-                    .collect_vec()
-        };
+
+        let protocol_fees = self
+            .fee_policies
+            .iter()
+            .filter_map(|fee_policy| {
+                let outside_market_price =
+                    boundary::is_order_outside_market_price(&order_, &quote_, order.data.kind);
+                match (outside_market_price, &fee_policy.order_class) {
+                    (_, OrderClass::Any) => Some(&fee_policy.policy),
+                    (true, OrderClass::Limit) => Some(&fee_policy.policy),
+                    (false, OrderClass::Market) => Some(&fee_policy.policy),
+                    _ => None,
+                }
+            })
+            .flat_map(|policy| match policy {
+                policy::Policy::Surplus(variant) => variant.apply(&order),
+                policy::Policy::PriceImprovement(variant) => variant.apply(&order, quote),
+                policy::Policy::Volume(variant) => variant.apply(&order),
+            })
+            .chain(partner_fee)
```

Comment:
After all, I am not sure if we should give partner fee a priority here or not. Let's keep it this way for now, although we might revaluate in the future.